### PR TITLE
Resolve Cypress to the version `9.5.2`

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
PENDING CI && PENDING REVIEW

### What does this PR accomplish?
- Resolves the Cypress to the specific (latest) version for all packages

### Explanation
After the initial upgrade to the version 9.5.1 and then later to the 9.5.2, there was still a Cypress version `6.4.0` left in the `yarn.lock`. That could sometimes cause problems [like this one](https://github.com/metabase/metabase/runs/5570849881?check_suite_focus=true).

The only way to get rid of that old version was to explicitly resolve Cypress to version 9.5.2

### How to test?
- Everything should be working after this and all e2e tests should be passing
- If you search for `6.4.0` in `yarn.lock`, you shouldn't find Cypress under that version

### CODE REVIEW

**Comment by ariya** [comment]
Running `yarn why cypress` gives the following:

```
$ yarn why cypress
yarn why v1.22.17
[1/4] Why do we have the module "cypress"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "cypress@9.5.2"
info Has been hoisted to "cypress"
info This module exists because it's specified in "devDependencies".
info Disk size without dependencies: "7.6MB"
info Disk size with unique dependencies: "21.66MB"
info Disk size with transitive dependencies: "39.89MB"
info Number of shared dependencies: 155
=> Found "@types/testing-library__cypress#cypress@6.4.0"
info This module exists because "@types#testing-library__cypress" depends on it.
info Disk size without dependencies: "9.24MB"
info Disk size with unique dependencies: "43.7MB"
info Disk size with transitive dependencies: "67.01MB"
info Number of shared dependencies: 152
=> Found "@testing-library/cypress#cypress@6.4.0"
info Reasons this module exists
   - "@testing-library#cypress#@types#testing-library__cypress" depends on it
   - Hoisted from "@testing-library#cypress#@types#testing-library__cypress#cypress"
info Disk size without dependencies: "9.24MB"
info Disk size with unique dependencies: "43.7MB"
info Disk size with transitive dependencies: "67.01MB"
info Number of shared dependencies: 152
Done in 2.65s.
```

Is there a chance to upgrade `@testing-library/cypress` first to see what happens?

In fact, I'm not even sure if we still need this `testing-library` for our Cypress tests. Do we even take advantage of it?

**Comment by nemanjaglumac** [comment]
> Is there a chance to upgrade `@testing-library/cypress` first to see what happens?

I started doing that and it breaks a lot of tests. That gave me an insight into some of the possible root causes for the failures. If you prefer that we first upgrade it, I can do it but it will take some time to go through the failed tests and to fix them.

> In fact, I'm not even sure if we still need this `testing-library` for our Cypress tests. Do we even take advantage of it?

We're using it basically in every single test and relying on it big time!
`findByText`, findByRole`, `findByPlaceholderText`,... those are NOT native Cypress commands. They are from the cypress testing library.

**Comment by ariya** [comment]
> I started doing that and it breaks a lot of tests. That gave me an insight into some of the possible root causes for the failures.

In that case, let's proceed with this PR first (force Cypress resolution) and attempt to fix `testing-library` as the follow-up.

**Review by ariya — APPROVED** [review]
:rocket:

**Comment by nemanjaglumac** [comment]
Upgrading testing-library got rid of one of those warnings from `yarn why cypress`. The remaining one was coming from the `@types/testing-library__cypress`. It was so subtle that I missed it at first.

It seems that their team removed it as a dependency in [v7.0.0](https://github.com/testing-library/cypress-testing-library/releases/tag/v7.0.0), which enabled me to remove it from our `package.json` ([take a look at this commit](https://github.com/metabase/metabase/pull/21016/commits/ca91462d38afda0ced3edc6c468aefebbda32ba8)).

That finally got rid of the Cypress version 6.4.0 from the `yarn.lock`. 

Long story short - if https://github.com/metabase/metabase/pull/21016 passes all tests, this PR will not be needed and we can safely close it.
